src/java/org/apache/cassandra/metrics/CassandraMetricsRegistry.java

```diff
@@ -250,6 +271,43 @@ private void removeAlias(MetricName name)
         if (mBeanServer.isRegistered(name.getMBeanName()))
             MBeanWrapper.instance.unregisterMBean(name.getMBeanName(), MBeanWrapper.OnException.IGNORE);
     }
+
+    /**
+     * Very infrequently used meters generate a linear amount of tick work based on how long it has been
+     * since the meter was last marked or read. On scales of a year this can be enough to cause the first request
+     * that needs to mark the meter to time out. Once a day read every meter to force them to run Meter.tickIfNecessary
+     * so we only ever run at most one day worth of tick work per meter in the request path.
+     *
+     * This can be removed if we ever upgrade and switch the default MovingAverage from EWMA to SlidingWindowTimeAverages
+     */
+    private void tickMeters()
+    {
+        List<Throwable> failures = new ArrayList<>();
+        int droppedFailures = 0;
+        for (Meter meter : getMeters().values())
+        {
+            try
+            {
+                meter.getOneMinuteRate();
+            }
+            catch (Throwable t)
+            {
+                if (failures.size() < 10)
+                    failures.add(t);
+                else
+                    droppedFailures++;
+            }
+        }
+        if (!failures.isEmpty())
+        {
+            Throwable failure = null;
+            for (Throwable t : failures)
+                failure = Throwables.merge(failure, t);
+            // To avoid the scheduled task being cancelled don't leak exceptions
+            // Runs only once a day so noise is not an issue
+            logger.error(String.format("Had error(s) attempting to tick meter. Dropped %d exceptions.", droppedFailures), failure);
```

Comment:
Should we use the `warn` (or `info`) type instead? I don't think a user can fix this error anyway.

Reply:
It's not something that is ever expected so if it does happen things could be pretty broken not just with this ticking functionality. Since it will only log once a day it seems more appropriate to make it an error so it's more visible and can get fixed.